docs/modules/ROOT/pages/libraries.adoc

```diff
@@ -784,3 +784,99 @@ DS.Util.mapToObject(payload, function(x) x.language.name, function(v) v.language
 };
 ------------------------
 
+## Regex
+
+The regular expressions library based on the https://github.com/google/re2j[Google RE2/J] implementation.
+
+### `regexFullMatch(string pattern, string input)`
+Matches the entire input against the pattern (anchored start and end). If there's no match, returns `null`. If there's a match, returns a JSON object which has the following structure.
+- `string` - the original input string;
+- `captures` - array of captured subgroups in the match, if any;
+- `namedCaptures` - map of named subgroups, if any;
+
+*Example*
+
+.DataSonnet map:
+------------------------
+DS.Regex.regexFullMatch(@'h(?P<mid>.*)o', 'hello');
+------------------------
+.Result
+------------------------
+{
+  "captures": [
+    "ell"
+  ],
+  "namedCaptures": {
+    "mid": "ell"
+  },
+  "string": "hello"
+}
+------------------------
+
+### `regexPartialMatch(string pattern, string input)`
```

Comment:
You'd talked about adding a new method related to this that returns an array of results, I think there'd still be a lot of use for that.